mod/01_rating/spec/set/right/downvotee_search_spec.rb

```diff
@@ -121,7 +121,16 @@
               "+company"=>{"content"=>"[[#{company.name}]]",:type_id=>Card::PointerID},
               "+value"=>{"content"=>"Na einai kalytero anthropo apo ton patera toy", :type_id=>Card::PhraseID},
               "+year"=>{"content"=>"2015", :type_id=>Card::PointerID},
-              "+Link"=>{:content=>"http://www.google.com/?q=fringepeter", "type_id"=>Card::PhraseID}
+              "+source"=>{
+                "subcards"=>{
+                  "new source"=>{
+                    "+Link"=>{
+                      "content"=>"http://www.google.com/?q=fringepeter",
+                       "type_id"=>Card::PhraseID
+                    }
+                  }
+                }
+              }
```

Comment:
> the new API was supposed to clean things up, but this PR adds a lot more than it removes.

The new API helps clean things up but as the houndci requires a lot of style changes so things are broken into lines to pass the houndci checking. 

>  And this now appears to skip the "+Link" handling that is still in the code, no?

This part is trying to simulate the request what will be sent from the browser.

If you remember, the metric value creation page renders the new source editor to let users input source information. Before the new subcard API, the rendering was just straight forward. For example, the link input filed's name was just `card[subcards][+Link][content]` but now the name now is `card[subcards][+source][subcards][new source][subcards][+Link][content]`.
from `form.rb`
```ruby
def subcard_input_names
    return '' if !form_root_format || form_root_format == self
    "#{@parent.subcard_input_names}[subcards][#{card.contextual_name}]"
  end
```
I have to put a name to the new source card while rendering otherwise the name will be `card[subcards][+source][subcards][][subcards][+Link][content]`  which is treated as an array but not a subcard.

`+Link` handling is not in the code anymore now. Please refer to `type/metric_value.rb#create_source`. Instead, a source card will be created based on the subcards in the `+source`.